source/GameWindow.cpp

```diff
@@ -82,7 +82,11 @@ bool GameWindow::Init(bool headless)
 
 	// When running the integration tests, don't create a window nor an OpenGL context.
 	if(headless)
+#if defined(__linux__) && !SDL_VERSION_ATLEAST(2, 0, 22)
+		setenv("SDL_VIDEODRIVER", "dummy", true);
```

Comment:
The setenv function wants an int, not a bool.

```suggestion
		setenv("SDL_VIDEODRIVER", "dummy", 1);
```

From man 3 setenv:

```
NAME
       setenv - change or add an environment variable

SYNOPSIS
       #include <stdlib.h>

       int setenv(const char *name, const char *value, int overwrite);

       int unsetenv(const char *name);
```

Reply:
There was no bool in C in the 1970s when setenv was introduced. Fortran, ever the avant-garde language, had its "logical" data type quite a while before that. However, setenv is a holdover from 1970s C + UNIX. It didn't import this advanced feature of Fortran during later C language versions.

Reply:
`true` converts to 1, and since the function wants a truthy/falsy value using an int is a bit weird imo

Reply:
Just use `SDL_TRUE` then, it was made for this exact situation.

Reply:
> true converts to 1, and since the function wants a truthy/falsy value using an int is a bit weird imo

UNIX programmers in the 1970s would have been appalled by that statement.

Reply:
> > true converts to 1, and since the function wants a truthy/falsy value using an int is a bit weird imo
> 
> UNIX programmers in the 1970s would have been appalled by that statement.

Are we UNIX programmers in the 90s?

Reply:
> Are we UNIX programmers in the 90s?

I don't know about you, but I live in the 2020s. The bold new era of C++20 and POSIX (UNIX) 1003.1-2017. Long live ISO/IEC 1539-1:2018!